pkg/service/service.go

```diff
@@ -68,14 +70,20 @@ func (svc *svcInfo) deepCopyToLBSVC() *lb.SVC {
 		backends[i] = *backend.DeepCopy()
 	}
 	return &lb.SVC{
-		Frontend:  *svc.frontend.DeepCopy(),
-		Backends:  backends,
-		Type:      svc.svcType,
-		Name:      svc.svcName,
-		Namespace: svc.svcNamespace,
+		Frontend:      *svc.frontend.DeepCopy(),
+		Backends:      backends,
+		Type:          svc.svcType,
+		TrafficPolicy: svc.svcTrafficPolicy,
+		Name:          svc.svcName,
+		Namespace:     svc.svcNamespace,
 	}
 }
 
+func (svc *svcInfo) requireNodeLocalBackends() bool {
+	return (svc.svcType == lb.SVCTypeNodePort || svc.svcType == lb.SVCTypeLoadBalancer) &&
```

Comment:
Shouldn't we add `lb.SVCTypeExternalIPs` to the list? Check https://msazure.club/kubernetes-services-and-iptables/ (a source of the truth when it comes to the kube-proxy semantics).

Reply:
Good catch. I was following https://github.com/kubernetes/kubernetes/blob/42fe74cd2c2d09a5fa80976bb82c91caa778738e/pkg/api/service/util.go#L71 but forgot that Cilium has additional types that K8s doesn't have. Will fix!